src/Sylius/Bundle/AdminBundle/Command/CreateAdminUserCommand.php

```diff
@@ -75,46 +71,84 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         return Command::INVALID;
     }
 
-    private function createEmailQuestion(): Question
+    private function askAdminUserData(): AdminUserInterface
     {
-        $question = new Question('Email');
-        $question->setValidator(function (?string $email) {
-            if (!filter_var($email, FILTER_VALIDATE_EMAIL) || $email === null) {
-                throw new \InvalidArgumentException('The e-mail address provided is invalid. Please try again.');
-            }
+        /** @var AdminUserInterface $adminUser */
+        $adminUser = $this->adminUserFactory->createNew();
 
-            return $email;
-        });
-        $question->setMaxAttempts(3);
+        $adminUser = $this->askEmail($adminUser);
+        $adminUser = $this->askUsername($adminUser);
+        $adminUser = $this->askFullName($adminUser);
+        $adminUser = $this->askPassword($adminUser);
 
-        return $question;
+        $locales = Locales::getNames();
+        $localeCode = $this->io->choice('Select the locale code', $locales, 'en_US');
+        $adminUser->setLocaleCode($localeCode);
+
+        $enabled = $this->io->confirm('Do you want to enable this admin user?', true);
+        $adminUser->setEnabled($enabled);
+
+        return $adminUser;
     }
 
-    private function adminUserExists(string $email): bool
+    private function askEmail(AdminUserInterface $adminUser): AdminUserInterface
     {
-        return null !== $this->adminUserRepository->findOneByEmail($email);
+        $email = $this->io->ask('Email');
+        $adminUser->setEmail($email);
+
+        $constraintViolationList = $this->validator->validateProperty($adminUser, 'email', ['sylius', 'sylius_user_create']);
+        if ($constraintViolationList->count() > 0) {
+            throw new \InvalidArgumentException('The email provided is invalid. Please try again.');
```

Comment:
Can't you just use the messages from the validator?

Reply:
It was my first idea. The problem is that not all current check messages match the CLI context, for example [this](https://github.com/Sylius/Sylius/blob/1.13/src/Sylius/Bundle/CoreBundle/Resources/translations/validators.en.yml#L130) one. We could standardize these names to make more sense in every context, wdyt?